src/foam/core/types.js

```diff
@@ -1091,3 +1091,23 @@ foam.CLASS({
     }
   ]
 });
+
+foam.CLASS({
+  package: 'foam.core',
+  name: 'FUIDProperty',
+  extends: 'Property',
+
+  properties: [
+    {
+      name: 'adapt',
+      value: function(_, a, p) {
+        var s = typeof a === 'number'   ? String(a)                :
+                a && a.toString         ? a.toString()             :
+                                          ''                       ;
+        return p.trim ? s.trim() : s;
```

Comment:
Why is this p.trim instead of s.trim?

Reply:
p is passed in as a param ... i assumed p.trim is a flag for whether the result should be trimmed, copied from StringPropertyInfo adapt

Reply:
Updated to s.trim()